scripts/check_style.sh

```diff
@@ -4,6 +4,16 @@ set -eu
 
 ERROR_LOG="$(mktemp -t check_style_XXXXXX.log)"
 
+if [ "$(uname)" == "Darwin" ]; then
+    if ! command -v ggrep &> /dev/null
+    then
+        brew install grep # install GNU grep on macOS
+    fi
+    grepCommand="ggrep"
+else
+    grepCommand="grep"
+fi
```

Comment:
A style check script should absolutely not modify your system by installing new software. If it requires something and it's not available, it just fail.

We actually added a portable `grep` helper recently, so please source `common.sh` and use it: https://github.com/ethereum/solidity/blob/fbbac9c1969c054071d9d6456e5151cb4ae53c19/scripts/common.sh#L317-L326